api/pull_request_test.go

```diff
@@ -2,42 +2,295 @@ package api
 
 import (
 	"encoding/json"
+	"fmt"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestPullRequest_ChecksStatus(t *testing.T) {
-	pr := PullRequest{}
+func TestChecksStatus_NoCheckRunsOrStatusContexts(t *testing.T) {
+	t.Parallel()
+
 	payload := `
+	{ "statusCheckRollup": { "nodes": [] } }
+	`
+	var pr PullRequest
+	require.NoError(t, json.Unmarshal([]byte(payload), &pr))
+
+	expectedChecksStatus := PullRequestChecksStatus{
+		Pending: 0,
+		Failing: 0,
+		Passing: 0,
+		Total:   0,
+	}
+	require.Equal(t, expectedChecksStatus, pr.ChecksStatus())
+}
+
+func TestChecksStatus_SummarisingCheckRuns(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                 string
+		payload              string
+		expectedChecksStatus PullRequestChecksStatus
+	}{
+		{
+			name:                 "QUEUED is treated as Pending",
+			payload:              singleCheckRunWithStatus("QUEUED"),
+			expectedChecksStatus: PullRequestChecksStatus{Pending: 1, Total: 1},
+		},
+		{
+			name:                 "IN_PROGRESS is treated as Pending",
+			payload:              singleCheckRunWithStatus("IN_PROGRESS"),
+			expectedChecksStatus: PullRequestChecksStatus{Pending: 1, Total: 1},
+		},
+		{
+			name:                 "WAITING is treated as Pending",
+			payload:              singleCheckRunWithStatus("WAITING"),
+			expectedChecksStatus: PullRequestChecksStatus{Pending: 1, Total: 1},
+		},
+		{
+			name:                 "PENDING is treated as Pending",
+			payload:              singleCheckRunWithStatus("PENDING"),
+			expectedChecksStatus: PullRequestChecksStatus{Pending: 1, Total: 1},
+		},
+		{
+			name:                 "REQUESTED is treated as Pending",
+			payload:              singleCheckRunWithStatus("REQUESTED"),
+			expectedChecksStatus: PullRequestChecksStatus{Pending: 1, Total: 1},
+		},
+		{
+			name:                 "COMPLETED / STARTUP_FAILURE is treated as Pending",
```

Comment:
Since this wasn't mentioned in the implementation as something we would expect under `PENDING`, I suspect it snuck in at some point (but haven't checked schema versions to validate this). Either way, perhaps we should treat it as a failure?

Reply:
Yes, good idea!

Reply:
I will do this in a follow up, no need to muddy the waters. I'm also interested in writing an issue that proposes we have a separate unknown state. I seem to have lost the tab but actions or some closely related project had an unknown state and I thought that made sense.